charts/datadog/templates/_helpers.tpl

```diff
@@ -914,4 +914,33 @@ Create RBACs for custom resources
   {{- end -}}
 {{- end -}}
 
+{{- define "process-checks-enabled" -}}
+  {{- if or .Values.datadog.processAgent.containerCollection .Values.datadog.processAgent.processCollection .Values.datadog.processAgent.processDiscovery .Values.datadog.apm.instrumentation.language_detection.enabled -}}
+    true
+  {{- else -}}
+    false
+  {{- end -}}
+{{- end -}}
 
+{{- define "should-enable-process-agent" -}}
+  {{- if or .Values.datadog.networkMonitoring.enabled .Values.datadog.serviceMonitoring.enabled -}}
+    true
+  {{- else if and (eq .Values.targetSystem "windows") (eq (include "process-checks-enabled" .) "true") -}}
+    true
+  {{- else if not .Values.agents.image.doNotCheckTag -}}
+    {{- $version := .Values.agents.image.tag | toString | trimSuffix "-jmx" -}}
+    {{- $length := len (split "." $version) -}}
+    {{- if and (eq $length 1) (eq $version "latest") -}}
+      {{- $version = "7.54.0" -}}
```

Comment:
Is the plan to update this value every time we release a new agent version? Is there some pre-existing variable we can use for this?

Reply:
I don't believe there is. The version is updated by changing the default value for `.Values.agents.image.tag` but users can override this. We only need to update the value to be a version recent enough for to be ahead of all version checks so it won't be often. I imagine one more time once we have the GA release of the run in core agent feature.

Reply:
maybe we could define a helper function that uses nearly the same logic as here: https://github.com/DataDog/helm-charts/blob/main/charts/datadog/templates/_helpers.tpl#L3-L20 (which uses a very outdated agent version..)

to return the `$version` value. what do you think?

Reply:
Would we still need to manually update `$version` it in that case?

Reply:
yes! but i figure it could be a function that can be reused and updated in just one spot

Reply:
Sounds good, I added it